src/convertToObject.js

```diff
@@ -3,16 +3,38 @@
 /**
  * Implement convertToObject function:
  *
- * Function takes a string with styles (see an example in [stylesString.js](./stylesString.js))
+ * Function takes a string with styles
+ * (see an example in [stylesString.js](./stylesString.js))
  * and returns an object where CSS properties are keys
- * and values are the values of related CSS properties (see an exampl in [test file](./convertToObject.test.js))
+ * and values are the values of related CSS properties
+ * (see an exampl in [test file](./convertToObject.test.js))
  *
  * @param {string} sourceString
  *
  * @return {object}
  */
 function convertToObject(sourceString) {
   // write your code here
+  const styled = {};
+  const parts = sourceString
+    .split(';');
+
+  parts.forEach(part => {
+    const colonIndex = part.indexOf(':');
+
+    if (colonIndex !== -1) {
+      const key = part
+        .slice(0, colonIndex)
+        .trim();
+      const value = part
+        .slice(colonIndex + 1)
+        .trim();
```

Comment:
Instead of using `if` and truncating from x to y at the start, you can use `replace` to remove spaces in the array. Then, split it into two parts using `split(':')`. The style of the value is separated by a colon. Finally, apply key-value parsing with `const [key, value] = str.trim().split(':')` to obtain the key and value.